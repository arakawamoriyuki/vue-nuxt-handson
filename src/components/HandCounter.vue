<template>
  <v-card
    class="mx-auto"
    max-width="200"
    outlined
  >
    <v-list-item three-line>
      <v-list-item-content>
        <div class="text-center text-h1">{{ count }}</div>
      </v-list-item-content>
    </v-list-item>
    <v-card-actions>
      <v-btn icon @click="decrement">
        <v-icon x-large>mdi-minus</v-icon>
      </v-btn>
      <v-spacer></v-spacer>
      <v-btn icon @click="increment">
        <v-icon x-large>mdi-plus</v-icon>
      </v-btn>
    </v-card-actions>
  </v-card>
</template>

<script>
import store from '@/store/HandCounterStore';

export default {
  name: 'hand-counter',
  store,
  components: {},
  computed: {
    count() {
      return this.$store.state.count;
    },
  },
  methods: {
    increment() {
      this.$store.dispatch('increment');
    },
    decrement() {
      this.$store.dispatch('decrement');
    },
  },
};
</script>
