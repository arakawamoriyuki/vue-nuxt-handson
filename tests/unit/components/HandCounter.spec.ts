import { shallowMount } from '@vue/test-utils';
import HandCounter from '@/components/HandCounter.vue';

describe('HandCounter', () => {
  it('HandCounterが表示される', () => {
    const wrapper = shallowMount(HandCounter);
    expect(wrapper).toMatchSnapshot();
  });
});
