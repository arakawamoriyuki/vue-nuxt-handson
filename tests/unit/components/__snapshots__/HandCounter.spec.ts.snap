// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandCounter HandCounterが表示される 1`] = `
<v-card-stub loaderheight="4" tag="div" maxwidth="200" outlined="true" class="mx-auto">
  <v-list-item-stub activeclass="" tag="div" threeline="true">
    <v-list-item-content-stub>
      <div class="text-center text-h1">0</div>
    </v-list-item-content-stub>
  </v-list-item-stub>
  <v-card-actions-stub>
    <v-btn-stub tag="button" activeclass="" icon="true" type="button">
      <v-icon-stub xlarge="" x-large="">mdi-minus</v-icon-stub>
    </v-btn-stub>
    <v-spacer-stub></v-spacer-stub>
    <v-btn-stub tag="button" activeclass="" icon="true" type="button">
      <v-icon-stub xlarge="" x-large="">mdi-plus</v-icon-stub>
    </v-btn-stub>
  </v-card-actions-stub>
</v-card-stub>
`;
